lib/runnable.js

```diff
@@ -343,11 +343,7 @@ Runnable.prototype.run = function(fn) {
 
     // allows skip() to be used in an explicit async context
     this.skip = function asyncSkip() {
-      done(new Pending('async skip call'));
-      // halt execution.  the Runnable will be marked pending
```

Comment:
per these comments, the intent was that the uncaught handler will ignore the thrown `Pending`.

for whatever reason (probably my fault), the uncaught handler is not ignoring the thrown `Pending`.

The proper fix then, IMO, is to modify `Runner#uncaught` instead:

```js
Runner.prototype.uncaught = function(err) {
  if (err instanceof Pending) {
    return;
  }
  /* etc */
}
```

Reply:
it also appears we need this on 390

```js
            if (self.test) {
              self.test.pending = true;
            }
```

Reply:
First part: done

Second part: IMO should not be necessary.

`self.test` is set when the `test` event is emitted, before any `beforeEach` hook.
The event order is:
- test start    <- `self.test` is set
- hook `beforeEach` start
- hook `beforeEach` end
- test end

Reply:
I'm not entirely sure how I got an error to be thrown there where `self.test` is `undefined`, but I'll see if I can reproduce the situation.